nix/pkgs/arvo-ropsten/builder.sh

```diff
@@ -0,0 +1,26 @@
+source $stdenv/setup
+
+cp -r $src tmp
+chmod -R u+w tmp
+
+ZUSE=tmp/sys/zuse.hoon
+AMES=tmp/sys/vane/ames.hoon
+ACME=tmp/app/acme.hoon
+
+# replace the mainnet azimuth contract with the ropsten contract
+sed --in-place \
+  's/\(\+\+  contracts  \)mainnet\-contracts/\1ropsten-contracts/' \
+  $ZUSE
+
+# increment the %ames protocol version
+sed -r --in-place \
+  's/^(=\+  protocol\-version=)([0-9])/echo "\1$(echo "\2+1" | bc)"/e' \
```

Comment:
This would rotate the ames version on every commit, right? I'm not sure we want to do that; the relevant conditions for a testnet is that a) the version doesn't match the livenet and b) the version doesn't match other testnets. If we want to automatically provisions networks from a given commit, I think we'll want to handle network isolation at the infrastructure level.

Reply:
Yeah the `arvo-ropsten` derivation is the same as `arvo`, except having a testnet configuration (with the protocol number incremented by one).

I'm interested in producing and maintaining a long-lived testnet, so "mainnet + 1" feels appropriate as a canonical protocol number for that.  For ephemeral/short-lived testnets we could simply then avoid both the mainnet and canonical testnet protocol numbers, managing that elsewhere as you recommend.

(This is a bit out-of-bounds -- I haven't made a proposal for a canonical testnet, so an ad-hoc UP might be useful after the fact.  Also, this will obviously need to be patched to avoid incrementing out of bounds, but it would do the trick for the time being.)

Reply:
(It may be better and more Nix-appropriate to apply a proper patch here as well, rather than hack at the source with sed.)